src/plugins/Relight.js

```diff
@@ -75,31 +75,90 @@ class Relight extends React.Component {
    * The onMouseMove method tracks the mouse coordinates over the RelightLightDirectionControl component to allow the
    * style of the component to change and indicate to the user which direction the light should be shining from.
    * The threeCanvasProps are updated in state to cause a re-render each time the mouse is moved whilst the button
-   * is pressed over the component, this updates the props passed to the Three canvas.
+   * is pressed over the component, this tracks the current rotation and flipped status of the canvas,
+   * this also updates the props passed to the Three canvas.
    * */
-  onMouseMove(event, id) {
+  onMouseMove(event, id, rotation) {
     const control = document.getElementById(id);
     const boundingBox = control.getBoundingClientRect();
 
+    let xMove;
+    let yMove;
+    let style;
+
     if (event.type === 'mousemove') {
       event.preventDefault();
-      this.mouseX = event.clientX - boundingBox.left;
-      this.mouseY = event.clientY - boundingBox.top;
+      xMove = event.clientX - boundingBox.left;
+      yMove = event.clientY - boundingBox.top;
     } else if (event.type === 'touchmove') {
       this.mouseDown = true;
-      this.mouseX = event.touches[0].clientX - boundingBox.left;
-      this.mouseY = event.touches[0].clientY - boundingBox.top;
+      xMove = event.touches[0].clientX - boundingBox.left;
+      yMove = event.touches[0].clientY - boundingBox.top;
+    }
+
+    // rotation is the total degrees the canvas has rotated, i.e. it stacks, we need to get this back
+    // to a relative number by using modulus...
+    const rotationModulus = rotation % 360;
+
+    switch (rotationModulus) {
+      case 0:
+        this.mouseX = xMove;
+        this.mouseY = yMove;
+        this.lightX = (this.mouseX / 100) * 2 - 1;
+        this.lightY = (this.mouseY / 100) * 2 - 1;
+        this.lightX = this.flipped ? -this.lightX : this.lightX;
+        style =
+          `radial-gradient(at ` +
+          this.mouseX +
+          `% ` +
+          this.mouseY +
+          `%, #ffffff, #000000)`;
+        break;
+      case -270:
+      case 90:
+        this.mouseX = yMove;
+        this.mouseY = xMove;
+        this.lightX = (this.mouseX / 100) * 2 - 1;
+        this.lightY = -((this.mouseY / 100) * 2 - 1);
+        this.lightY = this.flipped ? -this.lightY : this.lightY;
+        style =
+          `radial-gradient(at ` +
```

Comment:
Just ignorance on Javascript string formatting methods, `+` is the more traditional approach; I don't use this style of string formatting anywhere in the project at this point.  So I would raise that as a lower priority issue for another time.  Otherwise I am blocking an urgent fix over a style issue.  Happy to change to use this style though (in the entire project) because I think it's cleaner.

Reply:
But yeah I could use a template here.